### PR TITLE
Use modern CSS to simplify input checkboxs and radios. Also improves …

### LINKED ISSUE #2899: Remove the extra label used for single checkboxes. It is used to easy styling but is bad for accessibility.
It is a part of #2895 but I believe it needs a little bit of extra attention. 

On removing the extra label, facing the CSS issues with the input tag.

### PR DESCRIPTION
**Pull request by frjo**
…accessibility since we are now using standard markup.

Fixes #2899

### CODE REVIEW

File: hypha/static_src/src/sass/public/components/_form.scss [hunk 1]
```diff
@@ -455,81 +442,41 @@
...
-        & + label::before {
-            transition: background-color $transition;
-        }
-
-        &:checked + label::before {
-            background: url('./../../images/tick.svg') $color--dark-blue center no-repeat;
-            background-size: 12px;
-            border: 1px solid $color--dark-blue;
+        height: 1rem;
+        width: 1rem;
```
Comment (theskumar) [inline hypha/static_src/src/sass/public/components/_form.scss]: You can use `1em` to make the checkbox adjust to font size used in local context.

File: hypha/static_src/src/sass/public/components/_form.scss [hunk 2]
```diff
@@ -455,81 +442,41 @@
...
+        background-position: center;
+        background-repeat: no-repeat;
+        background-size: 12px;
```
Comment (theskumar) [inline hypha/static_src/src/sass/public/components/_form.scss]: You can use relative sizing for the background here. 12px will not adjust in the local context.

**Review by theskumar — CHANGES_REQUESTED** [review]

File: hypha/static_src/src/sass/public/components/_form.scss (hunk 1 above)
Comment (frjo) [reply hypha/static_src/src/sass/public/components/_form.scss]: Fixed now.

File: hypha/static_src/src/sass/public/components/_form.scss (hunk 2 above)
Comment (frjo) [reply hypha/static_src/src/sass/public/components/_form.scss]: Fixed now.

**Comment by frjo** [comment]
Added the apply styles as well now.

File: hypha/static_src/src/sass/public/components/_form.scss [hunk 3]
```diff
@@ -455,81 +442,41 @@
...
-        top: 5px;
-        left: 5px;
-        z-index: -1;
-        width: 1px;
-        height: 1px;
-    }
+        appearance: none;
+        accent-color: $color--dark-blue;
+        height: .7rem;
+        width: .7rem;
```
Comment (theskumar) [inline hypha/static_src/src/sass/public/components/_form.scss]: `s/rem/em` here, the rest look good to me.

File: hypha/static_src/src/sass/apply/components/_form.scss [hunk 4]
```diff
@@ -490,109 +472,41 @@
...
-        position: absolute;
-        top: 5px;
-        left: 5px;
-        z-index: -1;
-        width: 1px;
-        height: 1px;
+        appearance: none;
+        accent-color: $color--dark-blue;
+        height: .7rem;
+        width: .7rem;
```
Comment (theskumar) [inline hypha/static_src/src/sass/apply/components/_form.scss]: `s/rem/em` here, the rest look good to me.

File: hypha/static_src/src/sass/apply/components/_form.scss [hunk 5]
```diff
@@ -74,25 +74,11 @@
...
-                    order: -1;
-                    padding: 0;
-                }
-
-                &__help {
-                    order: -2;
-                    width: 100%;
+                    display: flex;
+                    align-items: center;
+                    gap: .5rem;
```
Comment (theskumar) [inline hypha/static_src/src/sass/apply/components/_form.scss]: `s/rem/em` here, the rest look good to me.

**Review by theskumar — APPROVED** [review]
suggest minor changes, rest look good to me! :)

**Review by theskumar — APPROVED** [review]

**Comment by sandeepsajan0** [comment]
It looks good, just a minor thing that I noticed is a little misalignment in checkbox label and input. 

image

**Comment by frjo** [comment]
Good catch. I think I found a fix for it now.

Skärmavbild 2023-05-16 kl  09 05 05

**Pull request merged**